diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/AnnotationNewLineRule.kt

```diff
@@ -0,0 +1,80 @@
+import com.pinterest.ktlint.core.ast.ElementType.ANNOTATION_ENTRY
+import com.pinterest.ktlint.core.ast.ElementType.CLASS
+import com.pinterest.ktlint.core.ast.ElementType.FUN
+import com.pinterest.ktlint.core.ast.ElementType.FUN_KEYWORD
+import com.pinterest.ktlint.core.ast.ElementType.MODIFIER_LIST
+import com.pinterest.ktlint.core.ast.ElementType.PRIMARY_CONSTRUCTOR
+import com.pinterest.ktlint.core.ast.ElementType.SECONDARY_CONSTRUCTOR
+import com.pinterest.ktlint.core.ast.isWhiteSpace
+import org.cqfn.diktat.common.config.rules.RulesConfig
+import org.cqfn.diktat.ruleset.constants.Warnings
+import org.cqfn.diktat.ruleset.utils.*
+import org.jetbrains.kotlin.com.intellij.lang.ASTNode
+import org.jetbrains.kotlin.com.intellij.psi.impl.source.tree.PsiWhiteSpaceImpl
+import java.lang.StringBuilder
+
+/**
+ * This rule makes each annotation applied to a class, method or constructor is on its own line. Except: if first annotation of constructor, class or method
+ */
+class AnnotationNewLineRule : Rule("annotation-new-line") {
+
+    private lateinit var configRules: List<RulesConfig>
+    private lateinit var emitWarn: ((offset: Int, errorMessage: String, canBeAutoCorrected: Boolean) -> Unit)
+    private var isFixMode: Boolean = false
+
+    override fun visit(node: ASTNode,
+                       autoCorrect: Boolean,
+                       params: KtLint.Params,
+                       emit: (offset: Int, errorMessage: String, canBeAutoCorrected: Boolean) -> Unit) {
+        configRules = params.getDiktatConfigRules()
+        emitWarn = emit
+        isFixMode = autoCorrect
+
+        when (node.elementType) {
+            CLASS, FUN, PRIMARY_CONSTRUCTOR, SECONDARY_CONSTRUCTOR -> checkAnnotation(node)
+        }
+    }
+
+
+    private fun checkAnnotation(node: ASTNode) {
+        if (node.hasChildOfType(MODIFIER_LIST)) {
+            val modList = node.findChildByType(MODIFIER_LIST)
+            fixAnnotation(modList!!)
+        }
+    }
+
+    private fun fixAnnotation(node: ASTNode) {
+        if (node.getAllChildrenWithType(ANNOTATION_ENTRY).size <= 1)
+            return
+
+        node.getAllChildrenWithType(ANNOTATION_ENTRY).forEach {
+            if (!it.isFollowedByNewline() || !it.isBeginByNewline())
+                deleteSpaces(it, !it.isFollowedByNewline(), !it.isBeginByNewline())
+        }
+
+    }
+
+    private fun deleteSpaces(node: ASTNode, rightSide: Boolean, leftSide: Boolean) {
+        Warnings.ANNOTATION_NEW_LINE.warnAndFix(configRules, emitWarn, isFixMode, "${node.text} not on a single line",
+                node.startOffset) {
+            if (rightSide) {
+                if (node.treeNext?.isWhiteSpace() == true) {
+                    node.removeChild(node.treeNext)
+                }
+                node.treeParent.addChild(PsiWhiteSpaceImpl("\n"), node.treeNext)
+            }
+
+            if(leftSide) {
+                if (node.treePrev?.isWhiteSpace() == true) {
```

Comment:
I think this branch will never be executed: white space before an annotation is handled when `deleteSpaces` is applied to the previous annotation, and for the very first annotation in `MODIFIER_LIST` `treePrev` should always be `null`.

Reply:
If you want to handle situation like
```kotlin
class Example @Anno @Anno2 constructor(val foo: Int)
```
and make first annotation start on a new line, you should check `WHITE_SPACE` before `treeParent`, I think.

Reply:
Right, you can handle fix this situation by adding a newline, but it will be more correct if you alter `WHITE_SPACE` node before parent if current `treePrev` is null.